core/services/synchronization/explorer_client_test.go

```diff
@@ -107,7 +107,7 @@ func TestWebSocketClient_Send_Binary(t *testing.T) {
 
 func TestWebSocketClient_Send_Unsupported(t *testing.T) {
 	wsserver, cleanup := cltest.NewEventWebSocketServer(t)
-	t.Cleanup(cleanup)
+	defer cleanup()
```

Comment:
This was the only different case among `NewEventWebSocketServer` uses. It looks like the cleanup code was trying to fail the test after it had already completed, since that is when `t.Cleanup` funcs are called. `defer` on the other hand executes during the test, when failing is still valid.
```
2021/08/27 12:40:18 send on explorer client received unsupported message type -1
2021/08/27 12:40:18 http: panic serving 127.0.0.1:61797: Fail in goroutine after TestWebSocketClient_Send_Unsupported has completed
```

Reply:
interesting - yeah we did have some problems with `t.Cleanup(..)` already (e.g. there is a potential of a deadlock if `mock.AssertExpectations` is used inside it. Good idea to stop using it altogether.